src/prometheus/ser.rs

```diff
@@ -0,0 +1,406 @@
+            String::from(_name) + &self.filter_label.primary.help.to_owned(),
+        );
+
+        let gauge_secondary_opt = Opts::new(
+            String::from(_name) + &self.filter_label.secondary.name.to_owned(),
+            String::from(_name) + &self.filter_label.secondary.help.to_owned(),
+        );
+        self.gauge_state = Some(GaugeState {
+            label_key: "".to_string(),
+            primary_value: "".to_string(),
+            secondary_value: "".to_string(),
+            gauge_primary_opt,
+            gauge_secondary_opt,
+            name: String::from(_name),
+        });
+
+        Ok(self)
+    }
+
+    fn serialize_struct_variant(
+        self,
+        name: &'static str,
+        _variant_index: u32,
+        variant: &'static str,
+        _len: usize,
+    ) -> Result<Self::SerializeStructVariant> {
+        Err(Error::UnsupportedValue {
+            kind: format!("Struct Variant ({name}::{variant})"),
+        })
+    }
+
+    fn collect_str<T: ?Sized>(self, value: &T) -> std::result::Result<Self::Ok, Self::Error>
+    where
+        T: std::fmt::Display,
+    {
+        self.serialize_str(&value.to_string())
+    }
+
+    fn is_human_readable(&self) -> bool {
+        true
+    }
+
+    fn serialize_i128(self, _value: i128) -> std::result::Result<Self::Ok, Self::Error> {
+        Err(Error::UnsupportedValue {
+            kind: "i128".to_string(),
+        })
+    }
+
+    fn serialize_u128(self, _value: u128) -> std::result::Result<Self::Ok, Self::Error> {
+        Err(Error::UnsupportedValue {
+            kind: "u128".to_string(),
+        })
+    }
+}
+
+impl<'a> ser::SerializeStruct for &'a mut Serializer {
+    type Ok = ();
+    type Error = Error;
+
+    fn serialize_field<T>(&mut self, key: &'static str, value: &T) -> Result<()>
+    where
+        T: ?Sized + Serialize,
+    {
+        let Some(gauge_state) = &mut self.gauge_state else {
+            return Err(Error::ExpectedStartStruct)
+        };
+        if self.filter_label.include.iter().any(|e| key.contains(e)) {
+            gauge_state.label_key = key.to_string();
+            value.serialize(&mut **self)?;
+        }
+        Ok(())
+    }
+
+    fn end(self) -> Result<()> {
+        let Some(gauge_state) = &self.gauge_state else {
+            return Err(Error::ExpectedStartStruct)
+        };
+        let Ok(gauge_hour) = Gauge::with_opts(gauge_state.gauge_primary_opt.clone()) else {
+            return Err(Error::Message("Can not create gauge_primary".to_string()))
+        };
+        let Ok(gauge_month) = Gauge::with_opts(gauge_state.gauge_secondary_opt.clone()) else {
+            return Err(Error::Message("Can not create gauge_secondary".to_string()))
+        };
+        gauge_hour.add(gauge_state.primary_value.parse::<f64>().unwrap_or_default());
+        gauge_month.add(
+            gauge_state
+                .secondary_value
+                .parse::<f64>()
+                .unwrap_or_default(),
+        );
```

Comment:
Do we need to add all const_labels before setting the value of the gauge ? 
If so, skip my comment.
If not, could we add the value of the gaguge during the serialization of the value ?

Reply:
Ok, nothing to do then